services/worker/src/worker/job_runners/split/descriptive_stats.py

```diff
@@ -0,0 +1,355 @@
+    }
+    numerical_features = {
+        feature_name: feature
+        for feature_name, feature in features.items()
+        if feature.get("_type") == "Value" and feature.get("dtype") in NUMERICAL_DTYPES
+    }
+    if not categorical_features and not numerical_features:
+        raise NoSupportedFeaturesError(
+            "No features for statistics computation found. Currently supported types are: "
+            f"{NUMERICAL_DTYPES} and ClassLabel. "
+        )
+
+    con = duckdb.connect(":memory:")  # we don't load data in local db file, use local parquet file instead
+    # configure duckdb extensions
+    if extensions_directory is not None:
+        con.sql(f"SET extension_directory='{extensions_directory}';")
+    con.sql("INSTALL httpfs")
+    con.sql("LOAD httpfs")
+    con.sql("SET enable_progress_bar=true;")
+
+    # store data as local parquet file for fast querying
+    local_parquet_path = (
+        Path(local_parquet_directory) / PARQUET_FILENAME if local_parquet_directory else Path(PARQUET_FILENAME)
+    )
+    logging.info(f"Copying remote data to a local parquet file {local_parquet_path}. ")
+    con.sql(f"COPY (SELECT * FROM read_parquet({parquet_files_urls})) TO '{local_parquet_path}' (FORMAT PARQUET);")
+
+    # compute for ClassLabels (we are sure that these are discrete categories)
+    if categorical_features:
+        logging.info("Compute statistics for categorical features")
+    for feature_name, feature in tqdm(categorical_features.items()):
+        logging.debug(f"Compute statistics for ClassLabel feature {feature_name}")
+        class_label_names = feature["names"]
+        cat_column_stats: CategoricalStatsItem = compute_categorical_stats(
+            con,
+            feature_name,
+            class_label_names=class_label_names,
+            n_samples=num_examples,
+            parquet_filename=local_parquet_path,
+        )
+        stats.append(
+            StatsPerColumnItem(
+                column_name=feature_name,
+                column_type="class_label",
+                column_dtype=None,  # should be some int?
+                column_stats=cat_column_stats,
+            )
+        )
+
+    if numerical_features:
+        logging.info("Compute min, max, mean, median, std, histogram for numerical features. ")
+    for feature_name, feature in tqdm(numerical_features.items()):
+        feature_dtype = feature["dtype"]
+        num_column_stats: NumericalStatsItem = compute_numerical_stats(
+            con,
+            feature_name,
+            parquet_filename=local_parquet_path,
+            n_bins=histogram_num_bins,
+            n_samples=num_examples,
+            dtype=feature_dtype,
+        )
+        stats.append(
+            StatsPerColumnItem(
+                column_name=feature_name,
+                column_type="float" if feature_dtype in FLOAT_DTYPES else "int",
```

Comment:
are "float" and "int" part of a new vocabulary that we're providing? If so, it should be well documented. Maybe related to the discussion in https://github.com/huggingface/datasets-server/issues/655 (having a "simpler" set of column types, because features, with type and dtype, is a bit complex)